app/models/profile.rb

```diff
@@ -71,8 +71,18 @@ def results(host)
   # rubocop:enable Metrics/MethodLength
 
   def score
-    return 1 if hosts.blank?
+    return 1 if hosts_in_inventory.blank?
 
-    (hosts.count { |host| compliant?(host) }).to_f / hosts.count
+    (hosts_in_inventory.count { |host| compliant?(host) }).to_f /
+      hosts_in_inventory.count
+  end
+
+  # Apparently fixtures relations are set using Array instead of
+  # ActiveRecord associations, so scopes do not work on tests with fixtures
+  # This method is meant to fix such cases.
+  def hosts_in_inventory
```

Comment:
It might be better to mock this method or the scope in the unit test and maybe add a test for `in_inventory` to verify it returns the correct result or simply verify the resulting SQL.

However, there must be a way to avoid having to treat fixtures differently.